File: 10. Subprogram/D. Faktorisasi Prima/d_faktorisasi_prima.py
def isFactor(N, x):
    if N % x == 0:
        return True
    return False

N = int(input())
i = 2
pos = 0
while N > 1:
    isPrime = False
    power = 0
    if isFactor(N, i):
        isPrime = True
        pos += 1
        while N % i == 0:
            N = N // i
            power += 1
        if pos > 1:
            print(" x ", end="")
        print(i, end="")
        if (power > 1):
            print("^", power, sep="", end="")
    i += 1

File: 10. Subprogram/D. Faktorisasi Prima/d_faktorisasi_prima.rs
fn is_factor(num: i32, x: i32) -> bool {
    if num % x == 0 {
        true
    } else {
        false
    }
}

fn main() {
    let mut reader = String::new();
    std::io::stdin().read_line(&mut reader).expect("");
    let mut n: i32 = reader.trim().parse().expect("");
    let (mut i, mut pos): (i32, i32) = (2, 0);
    while n > 1 {
        let mut power: i32 = 0;
        if is_factor(n, i) {
            pos += 1;
            while n % i == 0 {
                n /= i;
                power += 1;
            }
            if pos > 1 {
                print!(" x ");
            }
            print!("{}", i);
            if power > 1 {
                print!("^{}", power);
            }
        }
        i += 1;
    }
}
